aten/src/ATen/native/sparse/SparseTensor.cpp

```diff
@@ -214,9 +159,18 @@ SparseTensor new_with_tensor_and_size_sparse(const LongTensor& indices, const Te
   AT_CHECK(sizes.size() == sparseDims + denseDims, "number of dimensions must be sparseDims (", sparseDims, ") + denseDims (", denseDims, "), but got ", sizes);
 
   LongTensor max_indices = std::get</* values */ 0>(indices.max(/* dim */ 1, /* keepdim */ false));
-  auto max_indices_accessor = max_indices.accessor<int64_t, 1>();
+  LongTensor cpu_max_indices;
+  if (max_indices.is_cuda()) {
+    cpu_max_indices = at::CPU(kLong).tensor(max_indices.sizes());
+    cpu_max_indices.copy_(max_indices);
+  } else {
+    cpu_max_indices = max_indices;
+  }
+  auto cpu_max_indices_accessor = cpu_max_indices.accessor<int64_t, 1>();
   for (int64_t d = 0; d < sparseDims; d++) {
-    int64_t max_index_in_dim = max_indices_accessor[d];
+    // NB: This used to sync ndim times to access each entry; now we copy
```

Comment:
we can't do this in a kernel?

Reply:
I guess it's a matter of sync/async pref. If we do the error tests in kernel we won't find out about it until the async error code finally gets back to us. Do you really want it in a kernel?